### ISSUE TITLE
Histogram extents incorrect when overlaid

### ISSUE THREAD

**Issue by jlstevens** [ioam/holoviews#124]
Histograms are not being positioned correctly and the extents are wrong when overlaid. Consider:

```python
import holoviews as hv
import numpy as np
top_style = style=dict(facecolor='r', alpha=0.5)
d1 = np.random.normal(0,5, (100,100))
d2 = np.random.normal(0,25, (100,100))

h1 = hv.Histogram(np.histogram(d1))
h2 = hv.Histogram(np.histogram(d2))(style=top_style)
h1 + h2 + (h1 * h2)
```

The two histograms have two different means (5 and 25) but after overlaying they are shown as having the same mean! It might be that the extents are wrong because of this incorrect shifting...

**Comment by philippjfr** [comment]
Fixed as of commit bc869fbb8. Separately from this bug we should figure out how we want tick labels on Histograms to behave in general.

**Comment by jlstevens** [comment]
Ok, I'll file the tick labels as a separate issue.
